crates/tabby/src/worker.rs

```diff
@@ -63,19 +64,25 @@ async fn make_chat_route(context: WorkerContext, args: &WorkerArgs) -> Router {
     )
 }
 
-async fn make_completion_route(context: WorkerContext, args: &WorkerArgs) -> Router {
+async fn make_completion_route(context: WorkerContext, args: &WorkerArgs, metrics_handle: PrometheusHandle) -> Router {
     context.register(WorkerKind::Completion, args).await;
 
     let code = Arc::new(context.client.clone());
     let logger = Arc::new(context.client);
     let completion_state = Arc::new(
         create_completion_service(code, logger, &args.model, &args.device, args.parallelism).await,
     );
-
-    Router::new().route(
-        "/v1/completions",
-        routing::post(routes::completions).with_state(completion_state),
-    )
+    let metrics_handle = Arc::new(metrics_handle);
+
+    Router::new()
+        .route(
+            "/v1/completions",
+            routing::post(routes::completions).with_state(completion_state),
+        )
+        .route(
```

Comment:
this can be put in `main` line 102? (since both completion worker / chat worker should expose metrics)

Reply:
I'll look into it shortly, I'm getting up to speed on the codebase as a whole

Reply:
It seems much better now.  

I'm having some issues running Tabby from a fresh built binary if I specify a model, any advice?

For testing metrics on `serve`, I can test it by not specifying a model (endpoints fail but metrics works), but for `worker` commands, it expects a model and it fails to run.

For example:

`RUST_BACKTRACE=1 target/debug/tabby worker::completion --url test.com --model 'TabbyML/StarCoder-1B' --token test`


`thread 'main' panicked at crates/tabby-common/src/registry.rs:72:51:
called `Result::unwrap()` on an `Err` value: Error(TrackableError { kind: Other, cause: Some(Cause(Os { code: 2, kind: NotFound, message: "No such file or directory" })), history: History([Location { module_path: "serdeconv::convert_json", file: "/home/redacted/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serdeconv-0.4.1/src/convert_json.rs", line: 53, message: "" }]) })`

Reply:
the PR looks good - will test the worker for you.